nova/api/openstack/compute/server_groups.py

```diff
@@ -168,6 +168,10 @@ def delete(self, req, id):
     def index(self, req):
         """Returns a list of server groups."""
         context = req.environ['nova.context']
+        params = common.get_pagination_params(req)
+        offset = params.get('offset', 0)
+        limit = CONF.api.max_limit
+        limit = min(limit, params.get('limit') or limit)
```

Comment:
This probably would have been nicer as `get_limit_and_offset()` in `nova/api/openstack/common.py` because all the similar code lives there already.